### PR TITLE
Add 0.15.0 migration guide

### LINKED ISSUE #819: Starknet 0.11 migration guide
### Feature Request

A migration guide for starknet.py 0.15 exists, so there should be a section about `Starknet 0.11` changes

### PR DESCRIPTION
**Pull request by cptartur**
Closes #819 

## Introduced changes

- Added a migration guide for changes introduced in version 0.16.0 (also related to Starknet 0.11.0)

##

- [ ] This PR contains breaking changes

### CODE REVIEW

File: docs/migration_guide.rst [hunk 1]
```diff
@@ -1,6 +1,27 @@
...
+2. Removed support for making transactions with version 0.
+
+   - Removed ``Deploy`` transaction.
+   - Removed deprecated ``make_declare_tx``.
+
+3. Removed deprecated ``client`` argument from Contract's :meth:`~starknet_py.contract.Contract.__init__` and :meth:`~starknet_py.contract.Contract.from_address`.
+4. Removed deprecated ``net.l1`` L1<>L2 messaging module.
+
 **********************
 0.15.0 Migration guide
```
Comment (war-in) [inline docs/migration_guide.rst:26]: 0.15.0 is not released yet 😅

File: docs/migration_guide.rst [hunk 2]
```diff
@@ -1,6 +1,27 @@
...
+
+Deprecation
+-----------
+
+1. Deprecated Client's :meth:`~starknet_py.net.client.Client.net` property.
+
+Breaking changes
+----------------
+
+1. Removed deprecated ``AccountClient``
```
Comment (war-in) [inline docs/migration_guide.rst:16]: I once got a comment from someone that we do not have to write `deprecated` everywhere. 
Maybe let's write which class should be used instead?

**Review by war-in — CHANGES_REQUESTED** [review]

File: docs/migration_guide.rst (hunk 1 above)
Comment (cptartur) [reply docs/migration_guide.rst:29]: 🤦 Will combine 0.15.0 and 0.16.0 then

File: docs/migration_guide.rst [hunk 3]
```diff
@@ -6,6 +6,7 @@ Migration guide
 **********************
 
 0.15.0 makes the first step to remove the cairo-lang package as StarkNet.py dependency!
+It also adds initial support for Starknet 0.11.0 and related changes.
```
Comment (war-in) [inline docs/migration_guide.rst:9]: I would change the order of these two sentences. From the user's perspective, 0.11 is more important

**Comment by war-in** [comment]
Docs are failing. Could you resolve that?

**Comment by cptartur** [comment]
> Docs are failing. Could you resolve that?

Should be working now.

**Review by war-in — APPROVED** [review]

**Pull request merged**